### PR TITLE
[stable/openvpn] issue #14695 adding crl.pem to secret

### LINKED ISSUE #14824: stable/openvpn Upgrade fails due to lack of crl.pem
**Describe the bug**
After the merge of #14452, an upgrade of an existing installation fails with the following logs:
```
found existing certs - reusing  
Options error: --crl-verify fails with '/etc/openvpn/certs/crl.pem': No such file or directory
Options error: Please correct these errors.
Use --help for more information.                     
```
This is due to the fact that the crl is not generated by the startup script since certs already exist.

**Version of Helm and Kubernetes**:
helm v2.13.1
k8s v1.12

**Which chart**:
stable/openvpn

**What happened**:
Upgrade fails because crl.pem does not exist.

**What you expected to happen**:
Detect that crl.pem doesn't exist and generate it.

**How to reproduce it** (as minimally and precisely as possible):
Upgrade from a version before #14452 was merged to version after it was merged.

**Anything else we need to know**:
This can be fixed manually by generating the cert manually on an existing running pod before upgrading.
```
kubectl exec -it <pod-name> bash
cd /etc/openvpn/certs/
./easyrsa gen-crl
cp /etc/openvpn/certs/pki/crl.pem /etc/openvpn/certs/crl.pem
chmod 644 /etc/openvpn/certs/crl.pem
exit
```

### PR DESCRIPTION
**Pull request by irozet**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #14824, #14695

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/openvpn/README.md [hunk 1]
```diff
@@ -132,7 +132,10 @@ Certificates can be found in openvpn pod in the following files:
  `/etc/openvpn/certs/pki/ca.crt`
  `/etc/openvpn/certs/pki/issued/server.crt`
  `/etc/openvpn/certs/pki/dh.pem`
+ `/etc/openvpn/certs/pki/crl.pem`
 
 And optionally (see openvpn.taKey setting):
 
  `/etc/openvpn/certs/pki/ta.key`
+
+Note: using mounted secret makes creation of new client certificates impossible inside openvpn pod, since easyrca needs to write in certs directory, which is read-only.
```
Comment (jasongwartz) [inline stable/openvpn/README.md]: Nitpick: typo here, `easyrca` should be `easyrsa`

File: stable/openvpn/values.yaml [hunk 2]
```diff
@@ -76,7 +76,7 @@ openvpn:
   # keystoreSecret:
   # secret with openvpn certificates. If specified, certificates are taken from the secret
   # create secret with such command:
-  #    kubectl create secret generic openvpn-keystore-secret --from-file=./server.key --from-file=./ca.crt --from-file=./server.crt --from-file=./dh.pem
+  #    kubectl create secret generic openvpn-keystore-secret --from-file=./server.key --from-file=./ca.crt --from-file=./server.crt --from-file=./dh.pem --from-file=./crl.pem  [--from-file=./ta.key]
```
Comment (jasongwartz) [inline stable/openvpn/values.yaml]: Have you tested what happens if this file is not provided (doesn’t exist in the secret)? Is the crl mandatory?

Comment (irozet) [reply stable/openvpn/values.yaml]: I added a value openvpn.useCrl, to control whether to use/create crl.pem or not. By default it's 'false'. The patch also solves https://github.com/helm/charts/issues/14824

File: stable/openvpn/README.md (hunk 1 above)
Comment (irozet) [reply stable/openvpn/README.md]: fixed

File: stable/openvpn/values.yaml (hunk 2 above)
Comment (irozet) [reply stable/openvpn/values.yaml]: crl became mandatory in openvpn 3.12.5 (commit https://github.com/helm/charts/commit/100d5a156b00d06214bff8eba8b259cce2daa63c#diff-426bf77ebc46805831ad90724aa62549), due to crl-verify

Comment (jasongwartz) [reply stable/openvpn/values.yaml]: Ah great, I didn’t realise that had been introduced - thanks for fixing it!

**Comment by irozet** [comment]
@chrisz100 please check this PR

**Comment by jasongwartz** [comment]
@irozet thanks for the updates - I’ll finish reviewing it soon

**Review by chrisz100 — APPROVED** [review]

**Review by jasongwartz — APPROVED** [review]

**Comment by jasongwartz** [comment]
@irozet thanks for this fix!

**Pull request merged**